lib/Agrammon/DB/Dataset.pm6

```diff
@@ -329,16 +329,21 @@ class Agrammon::DB::Dataset does Agrammon::DB {
         }
     }
 
-    method delete-input($var-name) {
-        my $instance;
+    method delete-instance($variable-pattern, $instance) {
+        return unless $variable-pattern and $instance;
+        my $username = $!user.username;
 
-        my $var = $var-name;
-        if $var ~~ s/\[(.+)\]/[]/ {
-            $instance = $0;
-        }
+        self.with-db: -> $db {
+            my $ret = $db.query(q:to/SQL/, $username, $!name, $variable-pattern ~ '%', $instance);
+                DELETE FROM data_new
+                WHERE data_dataset=dataset_name2id($1,$2) AND data_var LIKE $3
+                                                        AND data_instance = $4
+                RETURNING data_val
+            SQL
 
-        return $instance ?? self!delete-instance-variable($var, $instance)
-                         !! self!delete-variable($var);
+            my $rows = $ret.rows;
+            return $rows if $rows;
```

Comment:
Also here, probably want to hand back `0` if nothing changed?

Reply:
Removed return value